seabird_chat_ingest.proto

```diff
@@ -0,0 +1,144 @@
+// ChatEvent to prevent namespace conflicts between the main seabird.proto file.
+
+// SendMessageChatRequest requests for a message to be sent to a given channel.
+// Either SuccessChatEvent or FailedChatEvent SHOULD be returned in response if
+// possible. This MUST NOT cause any chat events from the bot to appear in the
+// output event stream.
+message SendMessageChatRequest {
+  string channel_id = 1;
+  string message = 2;
+}
+
+// SendPrivateMessageChatRequest requests for a message to be sent to a given
+// user. Either SuccessChatEvent or FailedChatEvent SHOULD be returned in
+// response if possible. This MUST NOT cause any chat events from the bot to
+// appear in the output event stream.
+message SendPrivateMessageChatRequest {
+  string user_id = 1;
+  string message = 2;
+}
+
+// JoinChannelChatRequest requests for the bot to join a channel with the given
+// name. Either JoinChannelEvent or FailedChatEvent SHOULD be returned in
+// response if possible.
+message JoinChannelChatRequest {
+  string channel_name = 1;
+}
+
+// LeaveChannelChatRequest requests for the bot to leave the channel with the
+// given ID. Either LeaveChannelEvent or FailedChatEvent SHOULD be returned in
+// response if possible.
+message LeaveChannelChatRequest {
+  string channel_id = 1;
+}
+
+// UpdateChannelChatRequest requests for a the channel to be updated with the
+// given metadata. Either ChangeChannelEvent or FailedChatEvent SHOULD be returned
+// in response if possible.
+message UpdateChannelChatRequest {
+  string channel_id = 1;
+  string topic = 2;
+}
+
+// Each ChatRequest has an ID (which can be attached to outgoing events - only
+// the first outgoing event with this ID will be used as the reply) and an inner
+// message type. Having an ID allows us to route an event back to the requestor
+// without having the chat backend implement a service.
+message ChatRequest {
+  string id = 1;
+
+  oneof inner {
+    SendMessageChatRequest send_message = 2;
+    SendPrivateMessageChatRequest send_private_message = 3;
+    JoinChannelChatRequest join_channel = 4;
+    LeaveChannelChatRequest leave_channel = 5;
+    UpdateChannelChatRequest update_channel = 6;
+  }
+}
+
+// HelloChatEvent must be the first event sent by the chat backend. If a
+// different event is sent, the connection will be closed.
+message HelloChatEvent {
+  common.Identity identity = 1;
+
+  common.Backend service_type = 2;
+}
+
+// This is a marker event to respond to seabird-core that performing an action
+// has succeeded. Note that most requests have explicit associated Events, so
+// this should only be used in absence of a matching event.
+message SuccessChatEvent {}
+
+// This is a marker event to respond to seabird-core that performing an action
+// has failed. Note that an action which takes too long will also be marked as
+// failed automatically. The time alotted for an action is not static and may
+// change based on load.
```

Comment:
This bit should be called out above. Also, expressing that you have a deadline and then saying that the deadline changes makes this not a deadline IMO.

Reply:
Yeah, I didn't call it a deadline, I just wanted to specify that core will only wait for an unspecified amount of time before considering a request failed.

Reply:
That doesn't provide the chat backend with any information, though. That's sort of the same as saying "your request could be failed immediately if there's too much load", which is fine, but don't express it as a timeout, express it as backpressure.